src/ui/public/modals/__tests__/confirm_modal.js

```diff
@@ -0,0 +1,114 @@
+import angular from 'angular';
+import expect from 'expect.js';
+import ngMock from 'ng_mock';
+import _ from 'lodash';
+
+describe('ui/modals/confirm_modal', function () {
+  let confirmModal;
+  let $rootScope;
+
+  beforeEach(function () {
+    ngMock.module('kibana');
+    ngMock.inject(function ($injector) {
+      confirmModal = $injector.get('confirmModal');
+      $rootScope = $injector.get('$rootScope');
+    });
+  });
+
+  function findByDataTestSubj(dataTestSubj) {
+    return angular.element(document.body).find(`[data-test-subj=${dataTestSubj}]`);
+  }
+
+  afterEach(function () {
+    const confirmButton = findByDataTestSubj('confirmModalConfirmButton');
+    if (confirmButton) {
+      angular.element(confirmButton).click();
+    }
+  });
+
+  describe('throws an exception', function () {
+    it('when no custom confirm button passed', function () {
+      try {
+        confirmModal('hi', { onConfirm: _.noop });
+        expect(true).to.be(false);
+      } catch (error) {
+        expect(error).to.not.be(undefined);
+      }
```

Comment:
According to the [docs](https://github.com/Automattic/expect.js) we can make assertions about errors like this:

```javascript
expect(fn).to.throw(); // synonym of throwException
expect(fn).to.throwError(); // synonym of throwException
expect(fn).to.throwException(function (e) { // get the exception object
  expect(e).to.be.a(SyntaxError);
});
expect(fn).to.throwException(/matches the exception message/);
expect(fn2).to.not.throwException();
```

So we can rewrite this test to be:

```javascript
expect(() => confirmModal('hi', { onConfirm: _.noop })).to.throw("Error message here");
```